wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c

```diff
@@ -78,7 +79,8 @@ extern const resource_hnd_t wifi_firmware_image;
 extern const resource_hnd_t wifi_firmware_clm_blob;
 #endif
 
-unsigned char r_buffer[BLOCK_BUFFER_SIZE];
+// Need to be aligned as whd_bus_spi_protocol.c reads a uint32_t
+alignas(4) unsigned char r_buffer[BLOCK_BUFFER_SIZE];
```

Comment:
you can use CY_ALIGN (4) instead alignas(4), and replace `#include <stdalign.h> to `cy_utils.h`

Reply:
@npal-cy - Alas, that won't build for either the  or the Pico W:
```
In file included from /media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c:23:
/media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c:95:10: error: expected declaration specifiers or '...' before numeric constant
   95 | CY_ALIGN(4) unsigned char r_buffer[BLOCK_BUFFER_SIZE];
      |          ^
/media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/core-lib/include/cy_utils.h:146:43: note: in definition of macro 'CY_ALIGN'
  146 |     #define CY_ALIGN(align)     __ALIGNED(align)
      |                                           ^~~~~
/media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c: In function 'host_get_resource_block':
/media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c:234:12: error: 'r_buffer' undeclared (first use in this function); did you mean 'setbuffer'?
  234 |     memset(r_buffer, 0, block_size);
      |            ^~~~~~~~
      |            setbuffer
/media/sboylan/LaCie/zephyr-wifi/zephyr/modules/hal/infineon/wifi-host-driver/WiFi_Host_Driver/resources/resource_imp/whd_resources.c:234:12: note: each undeclared identifier is reported only once for each function it appears in
[19/47] Building C object zephyr/kernel/CMakeFiles/kernel.dir/work.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /media/sboylan/LaCie/zephyr/cy8-testing/build
```
It turns out that __ALIGNED, used to define CY_ALIGN, is not itself defined.

Reply:
hm... __ALIGNED is part of CMSIS, looks like cy_utils.h has missed cmsis_compiler.h...Can you try to add cmsis_compiler.h before cy_utils.h?

Reply:
I can't figure out where it's pulling cmsis_compiler.h from, but that did fix the problem.  I pushed that change to BitBucket.  See what you think!